boa_engine/src/bytecompiler.rs

```diff
@@ -456,13 +456,13 @@ impl<'b> ByteCompiler<'b> {
     }
 
     #[inline]
-    fn compile_access(node: &'_ Node) -> Access<'_> {
+    fn compile_access<'a>(&mut self, node: &'a Node) -> Option<Access<'a>> {
         match node {
-            Node::Identifier(name) => Access::Variable { name: name.sym() },
-            Node::GetConstField(node) => Access::ByName { node },
-            Node::GetField(node) => Access::ByValue { node },
-            Node::This => Access::This,
-            _ => unreachable!(),
+            Node::Identifier(name) => Some(Access::Variable { name: name.sym() }),
+            Node::GetConstField(node) => Some(Access::ByName { node }),
+            Node::GetField(node) => Some(Access::ByValue { node }),
+            Node::This => Some(Access::This),
+            _ => None,
```

Comment:
This Clippy warning was already fixed, but the merge undid the fix

Reply:
Ah hell, sorry. Fixing.